pgmax/fg/groups.py

```diff
@@ -556,21 +554,34 @@ class FactorGroup:
         variable_group: either a VariableGroup or - if the elements of more than one VariableGroup
             are connected to this FactorGroup - then a CompositeVariableGroup. This holds
             all the variables that are connected to this FactorGroup
-
-    Attributes:
-        _variables_to_factors: maps set of connected variables to the corresponding factors
+        variable_names_for_factors: A list of list of variable names, where each innermost element is the
+            name of a variable in variable_group. Each list within the outer list is taken to contain
+            the names of the variables connected to a factor.
     """
 
     variable_group: Union[CompositeVariableGroup, VariableGroup]
-    _variables_to_factors: Mapping[FrozenSet, nodes.Factor] = field(init=False)
+    variable_names_for_factors: Sequence[List]
 
-    def __post_init__(self) -> None:
-        """Initializes a tuple of all the factors contained within this FactorGroup."""
-        object.__setattr__(
-            self,
-            "_variables_to_factors",
-            MappingProxyType(self._get_variables_to_factors()),
-        )
+    def __post_init__(self):
+        object.__setattr__(self, "num_factors", len(self.variable_names_for_factors))
```

Comment:
If you do this you might want to add `num_factors: int = field(init=False)` to the class definition. Otherwise mypy will complain when you re-enable the check